internal/service/dtc/dtc_record_cname_resource_test.go

```diff
@@ -0,0 +1,392 @@
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDtcRecordCnameUseTtl(name, serverName, true),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDtcRecordCnameExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "use_ttl", "true"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccDtcRecordCnameUseTtl(name, serverName, false),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDtcRecordCnameExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "use_ttl", "false"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func testAccCheckDtcRecordCnameExists(ctx context.Context, resourceName string, v *dtc.DtcRecordCname) resource.TestCheckFunc {
+	// Verify the resource exists in the cloud
+	return func(state *terraform.State) error {
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+		apiRes, _, err := acctest.NIOSClient.DTCAPI.
+			DtcRecordCnameAPI.
+			Read(ctx, utils.ExtractResourceRef(rs.Primary.Attributes["ref"])).
+			ReturnFieldsPlus(readableAttributesForDtcRecordCname).
+			ReturnAsObject(1).
+			Execute()
+		if err != nil {
+			return err
+		}
+		if !apiRes.GetDtcRecordCnameResponseObjectAsResult.HasResult() {
+			return fmt.Errorf("expected result to be returned: %s", resourceName)
+		}
+		*v = apiRes.GetDtcRecordCnameResponseObjectAsResult.GetResult()
+		return nil
+	}
+}
+
+func testAccCheckDtcRecordCnameDestroy(ctx context.Context, v *dtc.DtcRecordCname) resource.TestCheckFunc {
+	// Verify the resource was destroyed
+	return func(state *terraform.State) error {
+		_, httpRes, err := acctest.NIOSClient.DTCAPI.
+			DtcRecordCnameAPI.
+			Read(ctx, utils.ExtractResourceRef(*v.Ref)).
+			ReturnAsObject(1).
+			ReturnFieldsPlus(readableAttributesForDtcRecordCname).
+			Execute()
+		if err != nil {
+			if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
+				// resource was deleted
+				return nil
+			}
+			return err
+		}
+		return errors.New("expected to be deleted")
+	}
+}
+
+func testAccCheckDtcRecordCnameDisappears(ctx context.Context, v *dtc.DtcRecordCname) resource.TestCheckFunc {
+	// Delete the resource externally to verify disappears test
+	return func(state *terraform.State) error {
+		_, err := acctest.NIOSClient.DTCAPI.
+			DtcRecordCnameAPI.
+			Delete(ctx, utils.ExtractResourceRef(*v.Ref)).
+			Execute()
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func testAccDtcRecordCnameBasicConfig(name, serverName string) string {
+	config := fmt.Sprintf(`
+	resource "nios_dtc_record_cname" "test" {
+	canonical = %q
+	dtc_server = nios_dtc_server.test.name
+}
```

Comment:
```suggestion
resource "nios_dtc_record_cname" "test" {
	canonical = %q
	dtc_server = nios_dtc_server.test.name
}
```
Please fix the indentation for all the test cases